src/libraries/System.Text.Json/tests/System.Text.Json.Tests/Utf8JsonWriterTests.cs

```diff
@@ -4208,6 +4208,112 @@ private static string GetCommentExpectedString(bool prettyPrint)
             return Encoding.UTF8.GetString(ms.ToArray());
         }
 
+        [Fact]
+        public void WriteCommentsInArrayAsIndentedJson()
+        {
+            const string expectedIndentedJson =
+@"/*Comment at start of doc*/
+[
+  /*Comment as first array item*/
+  []
+  /*Comment in the middle of array*/,
+  {}
+  /*Comment as the last array item*/
+]
+/*Comment at end of doc*/";
+
+            var options = new JsonWriterOptions { Indented = true, SkipValidation = false };
```

Comment:
Why do we need to skip validation here? IIRC comments are still allowed with validation enabled.

Reply:
It was enabling validation: `SkipValidation = false`.